civihr_employee_portal/civihr_employee_portal.module

```diff
@@ -7026,15 +7025,18 @@ function _block_anonymous_access() {
     // This is the URL for the modal that is displayed when the user
     // clicks on "Click here to request one from your HR administrator"
     'request_new_account/ajax',
-    // PCHR-1589 : The guest user should be able to access these urls
-    // to access default drupal login and rest password page specially for maintenance mode
-    'user',
-    'user/password',
-    'user/login',
   ];
 
-  if ($user_is_anonymous && !in_array($current_path, $allowed_paths)) {
-    drupal_add_http_header('Location', "$base_url/welcome-page");
+  if ($user_is_anonymous) {
+    if (!in_array($current_path, $allowed_paths) && substr($current_path, 0, 4) !== 'user') {
```

Comment:
Notice that user can get to "user" page also with "/index.php?q=user" URL. This rule also applies to any url (/index.php?q=dashboard for example). However I'm not sure if we're going to consider this in this case.